patches/tModLoader/Terraria/Initializers/AssetInitializer.cs.patch

```diff
@@ -37,12 +37,33 @@
  			services.AddService(typeof(AssetReaderCollection), assetReaderCollection);
  			services.AddService(typeof(IAssetRepository), provider);
  		}
-@@ -597,6 +_,8 @@
+@@ -597,6 +_,29 @@
  			Main.ResourceSetsManager.LoadContent(mode);
  			Main.MinimapFrameManagerInstance.LoadContent(mode);
  			Main.AchievementAdvisor.LoadContent();
 +
 +			UICommon.LoadTextures(); // Added by TML.
++		}
++
++		private static bool canaryFilesChecked;
++		internal static void CheckCanaryFiles() {
++			if (canaryFilesChecked)
++				return;
++
++			// These checks ensure the latest Terraria is installed.
```

Comment:
Is this necessary? I'd prefer a `File.Exists` in `InitTMLContentManager`, since we have the `vanillaContentFolder` there at the time, and it is mandatory